### PR TITLE
Update template spec for tuple -> AliasSeq renaming

### PR DESCRIPTION
**Pull request by jamadagni**
This is to be applied in tandem with https://github.com/D-Programming-Language/phobos/pull/3756

### CODE REVIEW

**Comment by greenify** [comment]
I already bumped into this issue - please update the docs :)

**Comment by adamdruppe** [comment]
I want someone else to eyeball it before I hit merge tho... and I walso want the old "Tuple" link to work, and a note of some sort saying like "Template sequence parameters were previously known as tuple parameters. They were renamed to avoid confusion with <link to tuple>the other tuple</a>"

File: spec/template.dd [hunk 1]
```diff
@@ -528,50 +528,50 @@ $(GNAME TemplateAliasParameterDefault):
 )
 
     $(P Alias parameters enable templates to be parameterized with
-        any type of D symbol, including global names, local names,
+        any type of D symbol, including type names, global names, local names,
```
Comment (ntrel) [inline spec/template.dd]: We should make it clear that built-in types can't be passed to an alias parameter.

File: spec/template.dd [hunk 2]
```diff
@@ -703,43 +703,45 @@ $(H3 Specialization)
...
 
     $(P If the last template parameter in the $(I TemplateParameterList)
-        is declared as a $(I TemplateTupleParameter),
+        is declared as a $(I TemplateSequenceParameter),
         it is a match with any trailing template arguments.
-        The sequence of arguments form a $(I Tuple).
-        A $(I Tuple) is not a type, an expression, or a symbol.
-        It is a sequence of any mix of types, expressions or symbols.
+        Such a sequence of arguments can be defined using the template
+        $(I AliasSeq) in the Phobos std.traits module and will thus henceforth
```
Comment (ntrel) [inline spec/template.dd]: Isn't it in `std.meta`?

**Comment by ntrel** [comment]
Other than that, looks good :+1:

**Comment by ntrel** [comment]
Fixes in jamadagni#1.

**Comment by jamadagni** [comment]
Hi I have an exam coming up and so slow to respond/work on this. Will do so next week. Thanks for your patience.

**Comment by ntrel** [comment]
Ping - merge my pull request?

**Comment by CyberShadow** [comment]
LOL. Request fulfilled.

**Comment by CyberShadow** [comment]
@ntrel I guess you should resubmit that against dlang.org master now :)

**Comment by ntrel** [comment]
I should have tagged **jamadagni** to avoid ambiguity ;-)

Resubmitted: https://github.com/dlang/dlang.org/pull/1283

**Pull request merged**